core/src/components/searchbar/searchbar.tsx

```diff
@@ -240,45 +250,86 @@ export class Searchbar implements ComponentInterface {
     return Promise.resolve(this.nativeInput!);
   }
 
+  /**
+   * Emits an `ionChange` event.
+   *
+   * This API should be called for user committed changes.
+   * This API should not be used for external value changes.
+   */
+  private emitValueChange() {
+    const { value } = this;
+    // Checks for both null and undefined values
+    const newValue = value == null ? value : value.toString();
+    // Emitting a value change should update the internal state for tracking the focused value
+    this.focusedValue = newValue;
+    this.ionChange.emit({ value: newValue });
+  }
+
   /**
    * Clears the input field and triggers the control change.
    */
-  private onClearInput = (shouldFocus?: boolean) => {
+  private onClearInput = async (shouldFocus?: boolean) => {
     this.ionClear.emit();
 
-    // setTimeout() fixes https://github.com/ionic-team/ionic/issues/7527
-    // wait for 4 frames
-    setTimeout(() => {
-      const value = this.getValue();
-      if (value !== '') {
-        this.value = '';
-        this.ionInput.emit();
-
-        /**
-         * When tapping clear button
-         * ensure input is focused after
-         * clearing input so users
-         * can quickly start typing.
-         */
-        if (shouldFocus && !this.focused) {
-          this.setFocus();
+    return new Promise((resolve) => {
+      // setTimeout() fixes https://github.com/ionic-team/ionic/issues/7527
+      // wait for 4 frames
+      setTimeout(() => {
+        const value = this.getValue();
+        if (value !== '') {
+          this.value = '';
+          this.ionInput.emit();
```

Comment:
Should this emit `null`? `ion-textarea` emits `null` with `clearOnEdit` and `ion-input` will after this change: https://github.com/ionic-team/ionic-framework/pull/26005

Reply:
Looks like not explicitly passing an event to `emit` already causes it to emit as `null`, but I can make it explicit 👍